torchdata/dataloader2/dataloader2.py

```diff
@@ -257,3 +314,17 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
             for adapter_fn in self.datapipe_adapter_fns:
                 self.datapipe = adapter_fn(self.datapipe)
         self._datapipe_before_reading_service_adapt = clone(self.datapipe)
+
+    def _pause(self):
+        self._is_paused = True
+        if hasattr(self.reading_service, "_pause"):
+            self.reading_service._pause()
+        else:
+            warnings.warn("ReadingService doesn't support pause.")
+
+    def resume(self):
```

Comment:
Should we make this private and only expose the API through `DataLoader2Iterator`?

Reply:
yeah, let's make it private for now then if you have any concern.